### PR TITLE
Functional done

### PR DESCRIPTION
**Pull request by ClementCaton**

### CODE REVIEW

File: docs/Functional.md [hunk 1]
```diff
@@ -1,51 +1,121 @@
...
 - The service will be able to ask the client to accept the use of his location.
 - The service will be able to take a postal address and return a time for Santa to arrive.
 - The service will be able to take a coordinate and return a time for Santa to arrive.
+- The service will take into consideration the [Equation of time](https://en.wikipedia.org/wiki/Equation_of_time).
 
 ### Non-functional requirements
 
 - The service will be able to handle 10k requests per minute.
 - The service will not call an external API.
+- The service will be scalable thanks to kubernetes
```
Comment (Guillaume-Riviere) [inline docs/Functional.md:30]: Kubernetes doesn't have to be in the functional, it's too technical

File: docs/Functional.md [hunk 2]
```diff
@@ -1,51 +1,121 @@
...
+## Personae
+
+<img src="./thispersondoesnotexist/dev2.jpg" alt="drawing" width="200"/><br>
+Name: Marc<br>
+Age: 22<br>
+Education: Master's degree in computer science<br>
+Background: Student<br>
+Skills: Good knowledge of computer science<br>
+Geography: France<br>
+Platform: Web Desktop<br>
```
Comment (Guillaume-Riviere) [inline docs/Functional.md:41]: What is web desktop ? Mozilla ? Chrome ? Internet Explorer ?

File: docs/Functional.md [hunk 3]
```diff
@@ -1,51 +1,121 @@
...
+Pain points or frustrations: I don't like to sign up to a service<br>
+
+<img src="./thispersondoesnotexist/kid.jpg" alt="drawing" width="200"/><br>
+Name: Lisa<br>
+Age: 8<br>
+Education: None<br>
+Background: Child<br>
+Skills: Knows of the existance of Santa<br>
+Geography: UK<br>
+Platform: Web Ipad<br>
```
Comment (Guillaume-Riviere) [inline docs/Functional.md:53]: Which size or format for the Ipad ?

File: docs/Functional.md [hunk 4]
```diff
@@ -1,51 +1,121 @@
...
+Goal: Find out when Santa will arrive<br>
+Motivation: Embush Santa to prank him<br>
+Pain points or frustrations: I don't like hard interfaces<br>
```
Comment (Guillaume-Riviere) [inline docs/Functional.md:56]: What is considered as a bad interface ?

File: docs/Functional.md [hunk 5]
```diff
@@ -1,51 +1,121 @@
...
+
+<img src="./thispersondoesnotexist/old.jpg" alt="drawing" width="200"/><br>
+Name: James<br>
+Age: 65<br>
+Education: Right's degree<br>
+Background: Retired<br>
+Skills: Knows how to navigate a web page as long as it's not too hard<br>
+Geography: USA<br>
```
Comment (Guillaume-Riviere) [inline docs/Functional.md:64]: California or Florida ? Lot of time shift between both states

File: docs/Functional.md [hunk 6]
```diff
@@ -1,51 +1,121 @@
...
+Platform: Web Desktop<br>
+Goal: Show to his grandsons when Santa will arrive<br>
+Motivation: I want to see the exitment in my grandsons' eyes<br>
+Pain points or frustrations: I can't navigate if the interface is not easy to use<br>
```
Comment (Guillaume-Riviere) [inline docs/Functional.md:68]: What is considered as easy to use ?

File: docs/Functional.md [hunk 7]
```diff
@@ -1,51 +1,121 @@
...
+
```
Comment (Guillaume-Riviere) [inline docs/Functional.md:70]: Missing Personae for someone using a Phone, maybe someone using a foldable phone.
Maybe specify when the personae will use the service, ( Hours before Santa, or Weeks before Santa )

File: docs/Functional.md [hunk 8]
```diff
@@ -1,51 +1,121 @@
...
 
 ## Objectives.
-<!--Objectives are written by product designers in response to the requirements. They describe in a more specific way what the product will look like. Objectives may describe architectures, protocols and standards to which the product will conform. Measurable objectives are those that set some criteria by which the end product can be judged. Measurability can be in terms of some index of customer satisfaction or in terms of capabilities and task times. Objectives must recognize time and resource constraints. The development schedule is often part or a corollary of the objectives.-->
 
 It will be a single web page with an input field. The user will be able to enter a postal address or a coordinate and the page will show the time for Santa to arrive on the page.
```
Comment (Guillaume-Riviere) [inline docs/Functional.md:73]: What append if he accept the GPS position ? Do the search bar doesn't show ?

File: docs/Functional.md [hunk 9]
```diff
@@ -1,51 +1,121 @@
...
 The service will be able to scale depending on the number of requests per minute.
 It will be supported by the majority of current browsers OSs and devices. 
 
 ## User documentation.
-<!--In general, all of the preceding documents (except the logic specification) are used as source material for the technical manuals and online information, such as help pages, that are prepared for the product's users.-->
 
 Write your adress in the field. And press enter.
```
Comment (Guillaume-Riviere) [inline docs/Functional.md:79]: What if i don't have Internet, Does it still works ? 
What about on Phone, I don't have enter Button. 

What if a accept GPS position is accepted ? Do i still have to enter an address and press enter ?

File: docs/Functional.md (hunk 1 above)
Comment (ClementCaton) [reply docs/Functional.md:30]: I'll delete this line

File: docs/Functional.md (hunk 2 above)
Comment (ClementCaton) [reply docs/Functional.md:41]: unrelevent, it should work for all of them

File: docs/Functional.md (hunk 3 above)
Comment (ClementCaton) [reply docs/Functional.md:53]: unrelevent, it should work for all of them

File: docs/Functional.md (hunk 4 above)
Comment (ClementCaton) [reply docs/Functional.md:56]: unrelevent, it's hard, not bad
hard as in bard to navigate

File: docs/Functional.md (hunk 5 above)
Comment (ClementCaton) [reply docs/Functional.md:64]: unrelevent, talking about time shift is for the technical

File: docs/Functional.md [hunk 10]
```diff
@@ -1,51 +1,121 @@
...
+Pain points or frustrations: I don't like hard interfaces<br>
+
+<img src="./thispersondoesnotexist/old.jpg" alt="drawing" width="200"/><br>
+Name: James<br>
+Age: 65<br>
+Education: Right's degree<br>
+Background: Retired<br>
+Skills: Knows how to navigate a web page as long as it's not too hard<br>
+Geography: USA<br>
+Platform: Web Desktop<br>
```
[the remark this answers is not in the document]
Comment (ClementCaton) [reply docs/Functional.md:65]: unrelevent, it should work for all of them

File: docs/Functional.md [hunk 11]
```diff
@@ -1,51 +1,121 @@
...
+We don't want to limit our users base to be in Vierzon. So we will not do any physical exposure in Vierzon.
+We prefer to keep it online and accessible to everyone.
+
 ## The final product.
-<!--Ideally, the final product is a complete implementation of the functional specification and design change requests, some of which may result from formal testing and beta testing.-->
 
-The final product can be found at [xmas.algosup.com](https://xmas.algosup.com/).
\ No newline at end of file
+The final product can be found at [xmas.algosup.com](https://xmas.algosup.com/).
+
```
Comment (Guillaume-Riviere) [inline docs/Functional.md:121]: What the final product will looks like ? ( Mock-up )

If someone is blind, is there something for that ?

File: docs/Functional.md (hunk 6 above)
Comment (ClementCaton) [reply docs/Functional.md:68]: I'll make it clear

File: docs/Functional.md (hunk 7 above)
Comment (ClementCaton) [reply docs/Functional.md:70]: the foldable phone user is unrelevent, it is the same as a small tablet
I'll take into account the rest though

**Review by Guillaume-Riviere — CHANGES_REQUESTED** [review]
Check comments for modifications requested

File: docs/Functional.md (hunk 11 above)
Comment (ClementCaton) [reply docs/Functional.md:121]: "If someone is blind, is there something for that ?"
stop trolling please

File: docs/Functional.md (hunk 2 above)
Comment (Guillaume-Riviere) [reply docs/Functional.md:41]: Maybe say somewhere that it will be compatible with all devices and size.

File: docs/Functional.md (hunk 3 above)
Comment (Guillaume-Riviere) [reply docs/Functional.md:53]: Responsive design isn't specified.

File: docs/Functional.md (hunk 9 above)
Comment (Guillaume-Riviere) [reply docs/Functional.md:79]: If internet disconnected right before I enter my address

File: docs/Functional.md (hunk 11 above)
Comment (Guillaume-Riviere) [reply docs/Functional.md:121]: For people Visually handicapped for example

**Comment by ClementCaton** [comment]
I changed accordingly to what the QA wanted

File: docs/Functional.md (hunk 9 above)
Comment (PaulMarisOUMary) [reply docs/Functional.md:82]: Is the website able to work only with cached data?

**Review by Guillaume-Riviere — CHANGES_REQUESTED** [review]
Still missing explanation on User Documentation part
Doesn't specify what append when GPS Position is validated, Does the user still need to enter adress 
What if the user what to test another adress after the first one, Do he have to refresh the page, or click somewhere

File: docs/Functional.md [hunk 12]
```diff
@@ -13,6 +13,20 @@ Clement Caton Last update : 18/11/22
...
 
+## Glossary
+
+Christmas is a religious holiday celebrated by Christians on December 25. It is a time for family and friends to get together and exchange gifts. It is also a time for people to give to those less fortunate.
+
+GPS location is a system that uses satellites to determine the precise location of a device on the Earth.
+
+Postal address is a unique combination of numbers and letters that identifies a specific location.
+
+Coodrinates are a set of numbers that identify a specific location on the Earth.
```
Comment (Guillaume-Riviere) [inline docs/Functional.md:24]: Coordinates*

**Review by Guillaume-Riviere — CHANGES_REQUESTED** [review]

**Review by PaulMarisOUMary — APPROVED** [review]
We're coming back for you (:

**Pull request merged**